packages/gatsby-plugin-catch-links/src/catch-links.js

```diff
@@ -152,10 +152,18 @@ export const routeThroughBrowserOrApp = (
 
   if (hashShouldBeFollowed(origin, destination)) return true
 
+  // Regex to test against exclude patterns
   if (pluginOptions.excludePattern) {
-    const excludeRegex = new RegExp(pluginOptions.excludePattern)
-    if (excludeRegex.test(destination.pathname)) {
-      return true
+    if (Array.isArray(pluginOptions.excludePattern)) {
```

Comment:
Have you considered to add a variable

```javascript
const patterns = Array.isArray(pluginOptions.excludePattern) ? pluginOptions.excludePattern : [pluginOptions.excludePattern]
```
?

This makes the if/else statement obsolete.